frontend/src/services/storage.js

```diff
@@ -0,0 +1,38 @@
+const tokenKeyName = 'auth.access_token';
+class Storage {
+    constructor(type = 'localStorage') {
+        this.store = window[type];
+    }
+
+    get(key) {
+        return this.store.getItem(key);
+    }
+
+    set(key, value) {
+        return this.store.setItem(key, value);
+    }
+
+    removeItem(key) {
+        this.store.removeItem(key);
+    }
+
+    getToken() {
+        return this.get(tokenKeyName);
+    }
+
+    setToken(token) {
+        return this.set(tokenKeyName, token);
+    }
+
+    hasToken() {
+        return !!this.get(tokenKeyName);
+    }
+
+    removeToken() {
+        return this.removeItem(tokenKeyName);
+    }
+}
+export const sessionStorage = new Storage('sessionStorage');
```

Comment:
add method that will check if localStorage is accessible and create storage with this "driver". I mean

isLocalStorageAccessible = () => try { window.localStorage.setItem('test-local-storage', 1); window.localStorage.removeItem('test-local-storage'); return true; } catch(e) { return false; }

Reply:
and then check let storage = new Storage(); if (!isLocalStorageAccessible()) { storage = new Storage('session'); } export default storage;